scrapy_zyte_api/_params.py

```diff
@@ -217,61 +224,117 @@ def _set_http_request_cookies_from_request(
+            }
+            if input_cookie.path_specified:
+                output_cookie["path"] = input_cookie.path
+            output_cookies.append(output_cookie)
+        if output_cookies:
+            api_params["requestCookies"] = output_cookies
+    else:
+        api_params.setdefault("experimental", {})
+        if "requestCookies" in api_params["experimental"]:
+            request_cookies = api_params["experimental"]["requestCookies"]
+            if request_cookies is False:
+                del api_params["experimental"]["requestCookies"]
+            elif not request_cookies and isinstance(request_cookies, list):
+                logger.warning(
+                    (
+                        "Request %(request)r is overriding automatic request "
+                        "cookie mapping by explicitly setting "
+                        "experimental.requestCookies to []. If this was your "
+                        "intention, please use False instead of []. Otherwise, "
+                        "stop defining experimental.requestCookies in your "
+                        "request to let automatic mapping work."
+                    ),
+                    {
+                        "request": request,
+                    },
+                )
+            return
+        output_cookies = []
+        input_cookies = _get_all_cookies(request, cookie_jars)
+        input_cookie_count = len(input_cookies)
+        if input_cookie_count > max_cookies:
+            logger.warning(
+                (
+                    "Request %(request)r would get %(count)r cookies, but request "
+                    "cookie automatic mapping is limited to %(max)r cookies "
+                    "(see the ZYTE_API_MAX_COOKIES setting), so only %(max)r "
+                    "cookies have been added to this request. To silence this "
+                    "warning, set the request cookies manually through the "
+                    "experimental.requestCookies Zyte API parameter instead. "
+                    "Alternatively, if Zyte API starts supporting more than "
+                    "%(max)r request cookies, update the ZYTE_API_MAX_COOKIES "
+                    "setting accordingly."
+                ),
+                {
+                    "request": request,
+                    "count": input_cookie_count,
+                    "max": max_cookies,
+                },
+            )
+            input_cookies = input_cookies[:max_cookies]
+        for input_cookie in input_cookies:
+            output_cookie = {
+                "name": input_cookie.name,
+                "value": input_cookie.value,
+                "domain": input_cookie.domain,
+            }
+            if input_cookie.path_specified:
+                output_cookie["path"] = input_cookie.path
+            output_cookies.append(output_cookie)
+        if output_cookies:
+            api_params["experimental"]["requestCookies"] = output_cookies
```

Comment:
I turned:

```
original code
```

into 

```
if not experimental:
    new code
else:
    original code
```

I am unsure whether or not this is better than in-lining the if statements.

Reply:
Maybe I should extract 2 separate functions?